targets/targets.json

```diff
@@ -7503,7 +7503,6 @@
         "core": "Cortex-M7",
         "supported_toolchains": [
             "GCC_ARM",
-            "IAR",
             "ARMC6"
```

Comment:
Does something break if you change this to `ARM` instead of `ARM6`? Try to build with Mbed-CLI and Mbed-tools.

Reply:
We'd better use "ARM" wherever possible in our code base. For example, Mbed CLI 2 doesn't `ARMC6`, though it may not reference this config here for now.

Reply:
These are settings for a separate Samsung target, I've changed it to "ARM" for the target this PR relates to. Should I change this as well?

Reply:
Ok, maybe leave it then since this PR is not related to that target.